iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -194,15 +194,15 @@ private async Task SendMessagesAsync(CancellationToken cancellationToken)
                 {
                     _logger.LogInformation($"Device sending message {++messageCount} to IoT Hub...");
 
-                    Message message = PrepareMessage(messageCount);
+                    (Message message, string payload) = PrepareMessage(messageCount);
                     while (true)
                     {
                         try
                         {
                             await s_deviceClient.SendEventAsync(message);
-                            _logger.LogInformation($"Sent message {messageCount} with data [{message.GetBytes()}]");
+                            _logger.LogInformation($"Sent message {messageCount} of {payload}");
```

Comment:
Can't read the message bytes again after having sent it (when it was read), so have `PrepareMessage` return the payload as well.